pkg/cmd/helm/release/release.go

```diff
@@ -540,29 +539,32 @@ func (o *Options) BuildAndPackage(chartDir string) error {
 		if err != nil {
 			return errors.Wrapf(err, "failed to load Chart.yaml")
 		}
-
-		for i, dependency := range chartDef.Dependencies {
-			log.Logger().Infof("Adding repository for dependency %s", dependency.Name)
-			if dependency.Repository != "" {
-				c := &cmdrunner.Command{
-					Dir:  chartDir,
-					Name: o.HelmBinary,
-					Args: []string{"repo", "add", strconv.Itoa(i), dependency.Repository},
-				}
-				_, err = o.CommandRunner(c)
-				if err != nil {
-					return errors.Wrapf(err, "failed to add repository")
+		if o.ChartOCI {
+			log.Logger().Info("Not adding dependency repository for OCI charts")
+		} else {
+			for i, dependency := range chartDef.Dependencies {
+				log.Logger().Infof("Adding  dependency %s", dependency.Name)
+				if dependency.Repository != "" {
```

Comment:
This looks wrong to me. Shouldn't you check if dependency.Repository starts with oci:// and then skip it?